source/docs/essentials/extend-the-theme.md

```diff
@@ -1,6 +1,129 @@
+Even though you could reuse pages and components individually in a totally different theme,
+most of the time you might find easier and faster to start with the base theme and
+iterate from there.
+
+**Theme override is the mechanism allowing to extend, adapt, rewrite almost everything contained in the base theme.** It is what allows you to:
+- upgrade Front-Commerce and benefit from the latest component improvements,
+- customize how an extension is displayed,
+- or create a slightly different theme for the Black Friday in confidence!
+
+People having an understanding of [layout overrides in Magento](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/layout-override.html), [child themes in Wordpress](https://developer.wordpress.org/themes/advanced-topics/child-themes/) / [Prestashop](https://devdocs.prestashop.com/1.7/themes/reference/template-inheritance/parent-child-feature/), [templates override in Drupal](https://www.drupal.org/docs/8/theming/twig/working-with-twig-templates), [themes in CakePHP](https://book.cakephp.org/3.0/en/views/themes.html) or other similar implementations should easily understand this mechanism in Front-Commerce.
+
+Your own theme will be located in its own folder and will use by default components from
+parent theme(s) — at least from Front-Commerce base theme.
+You would then copy the files you want to override in your theme folder by maintaining
+an identical file structure.
+Your component will be used instead of the base one anywhere across the application.
+
+Themes are part of a module’s web area. Hence you must follow theses steps:
+
+1. define the module in the base configuration file
+2. copy the file (`js`, `scss` or `gql`) you want to override in the `theme` folder of your module
+3. customize its content in your module directory
+
+## Define the module in the base configuration file
+
+First, we need to create a minimalist module with the following file structure:
+
+```
+my-module
+└── web
+    └── theme
+```
+
+Then, we need to tell Front-Commerce that this module exists in your [`.frontcommerce.js`](#TODO) file. If the module has been created at the root of your project, it would look like this:
+
+```diff
+module.exports = {
+  name: "Front Commerce",
+  url: "http://www.front-commerce.test",
+-  modules: [],
++  modules: ["./my-module"],
+  serverModules: [
+    // ...
+  ]
+};
+```
+
+Restart the application.
+That’s it! You have successfully registered the module in your Front-Commerce application.
+
+## Override a component
+
+Let's add the description of a `Product` to a `ProductItem` as an example of overriding the base theme.
+
+The original file is: `node_modules/front-commerce/src/web/theme/modules/ProductView/ProductItem/ProductItem.js`
+
+1. copy it to: `my-module/web/theme/modules/ProductView/ProductItem/ProductItem.js`
+2. add the description somewhere in your `ProductItem` with `{this.props.description}`.
+
+**However you will not see the description yet!**
```

Comment:
I understand that we want to avoid unnecessary restarts when overriding `ProductItem` but people might try to edit the component by adding a static string and it won't update.

```suggestion
**But you are not done yet!**
```